lib/data/kinetic/Kinetic.js

```diff
@@ -0,0 +1,360 @@
+import crypto from 'crypto';
+import EventEmitter from 'events';
+import net from 'net';
+import stream from 'stream';
+
+import { errors } from 'arsenal';
+import kineticlib from 'kinetic-js';
+
+const HEADER_SZ = 9;
+
+class Kinetic {
+
+    /**
+     * @constructor
+     * @param {Object.<string>[]} drives - the list of drives
+     * @returns {Kinetic} - to allow for a functional style.
+     */
+    constructor(drives) {
+        this._sockets = [];
+        this._drives = drives;
+        this._sockets = this._drives.map(drive => ({
+            drive,
+            clusterVersion: 0,
+            connectionID: 0,
+            sock: this._newSocket(),
+            _request: false,
+            _requestGet: false,
+            _events: new EventEmitter(),
+        }));
+        this._chunkTab = [];
+        this._count = 0;
+        this._lexiKey = '0';
+        this.connect();
+        return this;
+    }
+
+    /**
+     * Connect the sockets to the drives
+     * @return {undefined}
+     */
+    connect() {
+        this._sockets.forEach((socket, index) => {
+            socket.sock.on('connect', () => {
+                this._bindReadable(socket, index);
+            }).connect(socket.drive);
+        });
+    }
+
+    getIt() {
+        for (let i = 0; i < this._sockets.length; i++) {
+            if (this._sockets[i]._request === false
+                && this._sockets[i]._requestGet === false) {
+                return i;
+            }
+        }
+        this._sockets.push(
+            {
+                drive: { host: '127.0.0.1', port: '8123' },
```

Comment:
It seems a bit strange to have hardcoded values here. Am I wrong?

Also, nit: Weird indent.